python/pyspark/sql/tests.py

```diff
@@ -4550,6 +4550,32 @@ def test_timestamp_dst(self):
         self.assertPandasEqual(pdf, df_from_pandas.toPandas())
 
 
+@unittest.skipIf(
+    not _have_pandas or not _have_pyarrow,
+    _pandas_requirement_message or _pyarrow_requirement_message)
+class MaxResultArrowTests(unittest.TestCase):
+    # These tests are separate as 'spark.driver.maxResultSize' configuration
+    # is a static configuration to Spark context.
+
+    @classmethod
+    def setUpClass(cls):
+        cls.spark = SparkSession(SparkContext(
+            'local[4]', cls.__name__, conf=SparkConf().set("spark.driver.maxResultSize", "10k")))
```

Comment:
Okay, the last test failure looks weird and flaky (https://github.com/apache/spark/pull/25593#issuecomment-525286876). This test itself passed but seems like previously set `spark.driver.maxResultSize=10k` affects the other tests even though we stop the session and context explicitly.

This is fine for now in the master branch because this test is in a separate file and launched in a separate process; however, this is potentially an issue.

I am working around, in branch-2.4 specifically, by using `SparkSession(SparkContext(...))` for now since it's an orthogonal issue.

Reply:
are we going to do the same change for master branch later?

Reply:
We can although it's fine in the master for now as I described. Let me make a followup later to match.